### PR TITLE
Add support for f16 and f128 for toolchain upgrade to 6/28

### LINKED ISSUE #3303: Toolchain upgrade to nightly-2024-06-28 failed
Updating Rust toolchain from nightly-2024-06-27 to nightly-2024-06-28 requires source changes.
The failed automated run [can be found here.](https://github.com/model-checking/kani/actions/runs/9709572839)
Please review the changes at https://github.com/rust-lang/rust from https://github.com/rust-lang/rust/commit/4bc39f028d14c24b04dd17dc425432c6ec354536 up to https://github.com/rust-lang/rust/commit/9c3bc805dd9cb84019c124b9a50fdff1e62a7ec9. The log for this commit range is:
https://github.com/rust-lang/rust/commit/9c3bc805dd Auto merge of #127049 - flip1995:clippy-subtree-update, r=Manishearth
https://github.com/rust-lang/rust/commit/3ce7f9eb0c Merge commit '68a799aea9b65e2444fbecfe32217ce7d5a3604f' into clippy-subtree-update
https://github.com/rust-lang/rust/commit/68a799aea9 Auto merge of #12999 - flip1995:rustup, r=flip1995
https://github.com/rust-lang/rust/commit/585170ee60 Bump nightly version -> 2024-06-27
https://github.com/rust-lang/rust/commit/e9e7a815a7 Merge remote-tracking branch 'upstream/master' into rustup
https://github.com/rust-lang/rust/commit/4ddc8a2e4e Auto merge of #12992 - klensy:lintcheck-bump, r=Alexendoo
https://github.com/rust-lang/rust/commit/249595384b Auto merge of #126861 - GuillaumeGomez:migrate-run-make-invalid-library, r=Kobzol
https://github.com/rust-lang/rust/commit/036b38ced3 Auto merge of #126993 - petrochenkov:atvisord3, r=BoxyUwU
https://github.com/rust-lang/rust/commit/3394fe89d8 Add `ar` command in `run-make-support`
https://github.com/rust-lang/rust/commit/449cde32ad Migrate `run-make/invalid-library` to `rmake.rs`
https://github.com/rust-lang/rust/commit/84146b372a Add `ar_command` in `run_make_support`
https://github.com/rust-lang/rust/commit/aaaa9264dc Auto merge of #12995 - reillysiemens:fix-doc-markdown-devops-false-positive, r=blyxyas
https://github.com/rust-lang/rust/commit/a4ce33c0b2 Auto merge of #126728 - onur-ozkan:stage1-rustdoc, r=Kobzol
https://github.com/rust-lang/rust/commit/127fa2261b Auto merge of #127014 - jhpratt:rollup-45ic8f5, r=jhpratt
https://github.com/rust-lang/rust/commit/73016dc8a4 Rollup merge of #126983 - tgross35:f16-f128-smir, r=celinval
https://github.com/rust-lang/rust/commit/8905be5ef3 Rollup merge of #126980 - Borgerr:fix-extendfromslice-check, r=workingjubilee
https://github.com/rust-lang/rust/commit/d3debc0037 Rollup merge of #126929 - nnethercote:rm-__rust_force_expr, r=oli-obk
https://github.com/rust-lang/rust/commit/b1f43974c4 Rollup merge of #126928 - nnethercote:124141-pre, r=oli-obk
https://github.com/rust-lang/rust/commit/70b69a2384 Rollup merge of #126721 - Zalathar:nested-cov-attr, r=oli-obk
https://github.com/rust-lang/rust/commit/5ec93b8e36 Rollup merge of #126571 - nnethercote:less-maybe_whole-expr-2, r=petrochenkov
https://github.com/rust-lang/rust/commit/536235f07e Auto merge of #126907 - glaubitz:sparc-fixes, r=nagisa
https://github.com/rust-lang/rust/commit/1def498e3b Auto merge of #126692 - DianQK:nixos-patchelf, r=Nilstrieb
https://github.com/rust-lang/rust/commit/7033f9b14a Auto merge of #123918 - DianQK:clang-format, r=Kobzol
https://github.com/rust-lang/rust/commit/80b25b4c82 Fix doc_markdown DevOps false positive
https://github.com/rust-lang/rust/commit/f90d702e66 Auto merge of #120924 - xFrednet:rfc-2383-stabilization-party, r=Urgau,blyxyas
https://github.com/rust-lang/rust/commit/ba3f6812c1 ast: Standardize visiting order
https://github.com/rust-lang/rust/commit/c163d5c99d Add `cpp:fmt` to mingw-check-tidy
https://github.com/rust-lang/rust/commit/e17c16d55b Format C++ files in `llvm-wrapper`
https://github.com/rust-lang/rust/commit/de0ece2f29 Add clang-format to the external tool check
https://github.com/rust-lang/rust/commit/518b74ec5d Remove `f16` and `f128` ICE paths from smir
https://github.com/rust-lang/rust/commit/5aac24909c std: test a variety of ways to extend a Wtf8Buf
https://github.com/rust-lang/rust/commit/e5167fe7bd set self.is_known_utf8 to false in extend_from_slice
https://github.com/rust-lang/rust/commit/7f37f8af5f coverage: Allow `#[coverage(..)]` on `impl` and `mod`
https://github.com/rust-lang/rust/commit/3262611cc5 coverage: Apply `#[coverage(..)]` recursively to nested functions
https://github.com/rust-lang/rust/commit/457fda1701 coverage: Detach `#[coverage(..)]` from codegen attribute handling
https://github.com/rust-lang/rust/commit/9f1f6a3387 Add clang-format to requirements.txt in tidy
https://github.com/rust-lang/rust/commit/57c8f6f252 Add `.clang-format`
https://github.com/rust-lang/rust/commit/3bbec6aade `sudo CI=green` && Review changes <3
https://github.com/rust-lang/rust/commit/01b3c24bf5 Rollup merge of #126893 - dtolnay:prec, r=compiler-errors
https://github.com/rust-lang/rust/commit/1b4c281fe7 RFC 2383: Stabilize `lint_reasons` in Clippy :paperclips:
https://github.com/rust-lang/rust/commit/dfaa53fd58 Auto merge of #125741 - petrochenkov:atvisord, r=davidtwco
https://github.com/rust-lang/rust/commit/379b761462 Inline and remove `maybe_whole_expr!`.
https://github.com/rust-lang/rust/commit/9828e960ab Remove `__rust_force_expr`.
https://github.com/rust-lang/rust/commit/2e4d547d4a Extra panic cases.
https://github.com/rust-lang/rust/commit/bca5cd3a9d Extend `tests/ui/macros/nonterminal-matching.rs`.
https://github.com/rust-lang/rust/commit/e57bbb3779 rustc_target: Build sparc_unknown_linux_gnu with -mcpu=v9 and -m32
https://github.com/rust-lang/rust/commit/a194f42d21 rustc_target: Rewrite sparc_unknown_linux_gnu spec to use TargetOptions
https://github.com/rust-lang/rust/commit/8998ce24e0 Replace Deref bounds on Interner in favor of a SliceLike trait
https://github.com/rust-lang/rust/commit/a155c38989 Split out IntoIterator and non-Iterator constructors for AliasTy/AliasTerm/TraitRef/projection
https://github.com/rust-lang/rust/commit/8f7175c199 bootstrap: Link against libatomic on 32-bit SPARC
https://github.com/rust-lang/rust/commit/606ada193f bump strip-ansi-escapes
https://github.com/rust-lang/rust/commit/dc8ac3ec91 rustc_llvm: Link against libatomic on 32-bit SPARC
https://github.com/rust-lang/rust/commit/572ae3b227 rustc_data_structures: Use portable AtomicU64 on 32-bit SPARC
https://github.com/rust-lang/rust/commit/863179081e Auto merge of #12986 - Alexendoo:cache-lintcheck-bin, r=flip1995
https://github.com/rust-lang/rust/commit/8c718e5524 ast: Standardize visiting order for attributes and node IDs
https://github.com/rust-lang/rust/commit/35ec4eb354 Rename the 2 unambiguous precedence levels to PREC_UNAMBIGUOUS
https://github.com/rust-lang/rust/commit/aa30dd444b Fix a typo in a comment.
https://github.com/rust-lang/rust/commit/32374a196d Auto merge of #12930 - DaniPopes:missing-const-for-fn-suggestion, r=Jarcho
https://github.com/rust-lang/rust/commit/2194304b05 Cache lintcheck binary in ci
https://github.com/rust-lang/rust/commit/9628130541 Auto merge of #12985 - llogiq:fix-integration-test, r=Alexendoo
https://github.com/rust-lang/rust/commit/51ccad6986 use short message format in integration test
https://github.com/rust-lang/rust/commit/26c556dd63 Auto merge of #12965 - KisaragiEffective:resolve-invalid-paths-on-bool-then, r=blyxyas
https://github.com/rust-lang/rust/commit/58fc27f571 Rollup merge of #126723 - estebank:dot-dot-dot, r=Nadrieril
https://github.com/rust-lang/rust/commit/0ce07f61db Auto merge of #12961 - GuillaumeGomez:fix-manual_unwrap_or_default, r=Alexendoo
https://github.com/rust-lang/rust/commit/54b45f7f93 Fix incorrect suggestion for `manual_unwrap_or_default`
https://github.com/rust-lang/rust/commit/fe2fe7feed Auto merge of #12972 - Jarcho:span_ex, r=Manishearth
https://github.com/rust-lang/rust/commit/4baae5d8b3 Add new `Span` utils to avoid both allocating and compressing/decompressing spans.
https://github.com/rust-lang/rust/commit/f6661f5b9b StaticForeignItem and StaticItem are the same
https://github.com/rust-lang/rust/commit/3e84ca8ac9 Auto merge of #12368 - vohoanglong0107:unnecessary-min, r=Alexendoo
https://github.com/rust-lang/rust/commit/2f9f204123 feat: unnecessary_min_max lint
https://github.com/rust-lang/rust/commit/08fa5f0460 bootstrap-rustdoc: use current stage if download-rustc enabled
https://github.com/rust-lang/rust/commit/3baafd2e8c Fix `...` in multline code-skips in suggestions
https://github.com/rust-lang/rust/commit/9749d990ed resolve `clippy::invalid_paths` on `bool::then`
https://github.com/rust-lang/rust/commit/4aee08f999 Auto merge of #12963 - lochetti:fix_12874, r=Centri3
https://github.com/rust-lang/rust/commit/b147b6d03d Don't lint implicit_return on proc macros
https://github.com/rust-lang/rust/commit/c693f31ee2 Update float tests to include `f16` and `f128`
https://github.com/rust-lang/rust/commit/8cde354f0b Resolve Clippy `f16` and `f128` `unimplemented!`/`FIXME`s
https://github.com/rust-lang/rust/commit/80e43c128c patch an ELF file using `--add-rpath`
https://github.com/rust-lang/rust/commit/4281c2c3d5 patch `rust-lld` and `ld.lld` on NixOS
https://github.com/rust-lang/rust/commit/29cc5c691c Auto merge of #12942 - Jarcho:ex_proc_macro, r=Manishearth
https://github.com/rust-lang/rust/commit/4b7ae63fbf Use a dedicated type instead of a reference for the diagnostic context
https://github.com/rust-lang/rust/commit/bd75e44f6e Auto merge of #12655 - SpencerAWill:Add-applicability-filter, r=xFrednet
https://github.com/rust-lang/rust/commit/7e1ed1ad71 Auto merge of #12953 - Centri3:back, r=Centri3
https://github.com/rust-lang/rust/commit/e28b998fe6 Add myself back to reviewer rotation
https://github.com/rust-lang/rust/commit/7218fdd2db Fix other tools
https://github.com/rust-lang/rust/commit/61fc1aec74 Rework precise capturing syntax
https://github.com/rust-lang/rust/commit/9e54ff2952 Auto merge of #12906 - lochetti:manual_unwrap_or_if_let, r=y21
https://github.com/rust-lang/rust/commit/51c5eee9e6 Auto merge of #12945 - Jarcho:octal_escape, r=Alexendoo
https://github.com/rust-lang/rust/commit/0625183e28 Auto merge of #12903 - Jarcho:issue_12284, r=y21
https://github.com/rust-lang/rust/commit/9550481056 Auto merge of #12947 - xFrednet:00000-summer-assignments, r=xFrednet
https://github.com/rust-lang/rust/commit/198bbf87a0 Pause assignments to @xFrednet for summer break :beach_umbrella:
https://github.com/rust-lang/rust/commit/2ad53f4e38 Auto merge of #12943 - Jarcho:range_sym, r=llogiq
https://github.com/rust-lang/rust/commit/6172178d22 Auto merge of #12938 - elijah-potter:patch-1, r=flip1995
https://github.com/rust-lang/rust/commit/4b16e265a7 Rework `octal_escapes`.
https://github.com/rust-lang/rust/commit/f09650b347 Use symbols when raising range expressions.
https://github.com/rust-lang/rust/commit/e18b310874 Add more types to `is_from_proc_macro`
https://github.com/rust-lang/rust/commit/16efa56503 Auto merge of #12287 - Jarcho:issue_12250, r=llogiq
https://github.com/rust-lang/rust/commit/22710f33a8 Add lint `manual_inspect`
https://github.com/rust-lang/rust/commit/a002f93e51 `expr_use_ctxt` changes: * Delay the parsing of the use node * Mark when the `SyntaxContext` changes rather than return `None` * Return a default value if the HIR tree is broken rather than `None`
https://github.com/rust-lang/rust/commit/9f5d60f160 Auto merge of #12893 - kyleoneill:field_scoped_visibility_modifiers, r=blyxyas
https://github.com/rust-lang/rust/commit/3405ce3bca Add field_scoped_visibility_modifiers lint
https://github.com/rust-lang/rust/commit/8065e0f61c Auto merge of #12939 - Alexendoo:lintcheck-popular-crates, r=xFrednet
https://github.com/rust-lang/rust/commit/3a983c399a Merge lintcheck popular-crates bin as a subcommand
https://github.com/rust-lang/rust/commit/a2c9782128 Auto merge of #10398 - Alexendoo:auto-lintcheck, r=xFrednet
https://github.com/rust-lang/rust/commit/82345c30ae Auto merge of #12937 - AurelienFT:add_msrv_manual_patter_char_comparison, r=Alexendoo
https://github.com/rust-lang/rust/commit/bcc7b0e700 Fix minor typo
https://github.com/rust-lang/rust/commit/51c6630d4f Change MSRV check for manual_pattern_char_comparison only for pattern arrays
https://github.com/rust-lang/rust/commit/63388cbab8 add MSRV for manual_pattern_char_comparison
https://github.com/rust-lang/rust/commit/0dc265ff82 Auto merge of #12756 - y21:assigning_clones_lifetimes, r=Alexendoo
https://github.com/rust-lang/rust/commit/feb0671893 Add lintcheck diff github action
https://github.com/rust-lang/rust/commit/477b0c6a78 lintcheck: Add JSON output, diff subcommand
https://github.com/rust-lang/rust/commit/73c1bfbc57 Auto merge of #12880 - bitfield:fix_doc_nits_b, r=Alexendoo
https://github.com/rust-lang/rust/commit/82f0dc95a0 [`missing_const_for_fn`]: add machine-applicable suggestion
https://github.com/rust-lang/rust/commit/fa9274c99b Tweak output of import suggestions
https://github.com/rust-lang/rust/commit/ba347f9815 Auto merge of #126398 - flip1995:clippy-subtree-update, r=Manishearth
https://github.com/rust-lang/rust/commit/3bff119f63 Merge commit '3e5a02b13b1244545454752c6629b767522a44b1' into clippy-subtree-update
https://github.com/rust-lang/rust/commit/c3d463333f Fixup clippy tests
https://github.com/rust-lang/rust/commit/70ca9a1e7e Lint `manual_unwrap_or` for it let cases
https://github.com/rust-lang/rust/commit/059eaf1386 Fix ICE in `upper_case_acronyms`  and remove most of the string allocations.
https://github.com/rust-lang/rust/commit/7c86db4ea5 [ mut_range_bound ]: fix doc nits
https://github.com/rust-lang/rust/commit/8da5d64669 [ allow_attributes_without_reason ]: fix doc nits
https://github.com/rust-lang/rust/commit/35d284fcbf [ allow_attributes ]: fix doc nits
https://github.com/rust-lang/rust/commit/7ab4af336c [ needless_for_each ]: fix doc nits
https://github.com/rust-lang/rust/commit/5e60afb6cc [ arc_with_non_send_sync ]: fix doc nits
https://github.com/rust-lang/rust/commit/60508f546a deal with non-`Drop` types and add a test case for projections
https://github.com/rust-lang/rust/commit/20d1bb1867 [`assigning_clones`]: bail out when the source borrows from target
https://github.com/rust-lang/rust/commit/dff9164ac5 consider `copy_deref` a possible borrower
https://github.com/rust-lang/rust/commit/c8a7e6e5e6 Remove debug helper.
https://github.com/rust-lang/rust/commit/e4ce248af4 Complete filter functionality. Formatted upper filters to take more visual column space. Force filter wrapping with flex for small screens.
https://github.com/rust-lang/rust/commit/0b43e03ca9 Add applicability visually

### PR DESCRIPTION
**Pull request by jaisnan**
Adds support for f16 and f128, i.e 
1. adding translation to `irep` .
2. generating arbitrary values for the new types
3. generating basic invariants (checking if safe) for new types
4. Adds sanity testing for arbitrary on the new types.

Resolves #3303 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: tests/perf/s2n-quic
Comment (zhassan-aws) [inline tests/perf/s2n-quic:1]: I think the update here is unintentional.

File: cprover_bindings/src/goto_program/expr.rs [hunk 1]
```diff
@@ -98,7 +98,11 @@ pub enum ExprValue {
     // {}
     EmptyUnion,
     /// `1.0f`
+    HalfConstant(f16),
```
Comment (zhassan-aws) [inline cprover_bindings/src/goto_program/expr.rs:101]: I suggest adding `Float` to the name, e.g. `HalfFloatConstant` or `Float16Constant`.

File: cprover_bindings/src/goto_program/expr.rs [hunk 2]
```diff
@@ -581,6 +585,28 @@ impl Expr {
...
+    pub fn float16_constant(c: f16) -> Self {
+        expr!(HalfConstant(c), Type::float16())
+    }
+
+    /// `3.14159265358979323846264338327950288L`
+    pub fn float128_constant(c: f128) -> Self {
+        expr!(Float128Constant(c), Type::float128())
+    }
+
+    /// `union {_Float16 f; uint16_t bp} u = {.bp = 0x1234}; >>> u.f <<<`
```
Comment (zhassan-aws) [inline cprover_bindings/src/goto_program/expr.rs:598]: What is this comment saying?

File: cprover_bindings/src/goto_program/expr.rs [hunk 3]
```diff
@@ -581,6 +585,28 @@ impl Expr {
...
+    pub fn float16_constant_from_bitpattern(bp: u16) -> Self {
+        let c = f16::from_bits(bp);
+        Self::float16_constant(c)
+    }
+
+    /// `union {_Float128 f; __uint128_t bp} u = {.bp = 0x1234}; >>> u.f <<<`
```
Comment (zhassan-aws) [inline cprover_bindings/src/goto_program/expr.rs:604]: Same question for this one.

File: cprover_bindings/src/irep/to_irep.rs [hunk 4]
```diff
@@ -695,6 +717,24 @@ impl ToIrep for Type {
                     (IrepId::CCType, Irep::just_id(IrepId::Float)),
                 ],
             },
+            Type::Float16 => Irep {
+                id: IrepId::Floatbv,
+                sub: vec![],
+                named_sub: linear_map![
+                    (IrepId::F, Irep::just_int_id(10)),
```
Comment (zhassan-aws) [inline cprover_bindings/src/irep/to_irep.rs:724]: What does 10 represent?

File: cprover_bindings/src/irep/to_irep.rs [hunk 5]
```diff
@@ -695,6 +717,24 @@ impl ToIrep for Type {
...
+                    (IrepId::Width, Irep::just_int_id(16)),
+                    (IrepId::CCType, Irep::just_id(IrepId::Float16)),
+                ],
+            },
+            Type::Float128 => Irep {
+                id: IrepId::Floatbv,
+                sub: vec![],
+                named_sub: linear_map![
+                    (IrepId::F, Irep::just_int_id(112)),
```
Comment (zhassan-aws) [inline cprover_bindings/src/irep/to_irep.rs:733]: What does 112 represent?

File: cprover_bindings/src/irep/to_irep.rs [hunk 6]
```diff
@@ -254,6 +254,28 @@ impl ToIrep for ExprValue {
...
                 }
             }
+            ExprValue::HalfConstant(i) => {
+                let c: u16 = i.to_bits();
+                Irep {
+                    id: IrepId::Constant,
+                    sub: vec![],
+                    named_sub: linear_map![(
+                        IrepId::Value,
+                        Irep::just_bitpattern_id(c, mm.float_width, false)
```
Comment (zhassan-aws) [inline cprover_bindings/src/irep/to_irep.rs:264]: Should the width be 16 instead of `mm.float_width`?

File: cprover_bindings/src/irep/to_irep.rs [hunk 7]
```diff
@@ -254,6 +254,28 @@ impl ToIrep for ExprValue {
...
+                }
+            }
+            ExprValue::Float128Constant(i) => {
+                let c: u128 = i.to_bits();
+                Irep {
+                    id: IrepId::Constant,
+                    sub: vec![],
+                    named_sub: linear_map![(
+                        IrepId::Value,
+                        Irep::just_bitpattern_id(c, mm.float_width, false)
```
Comment (zhassan-aws) [inline cprover_bindings/src/irep/to_irep.rs:275]: Same here.

**Review by zhassan-aws — COMMENTED** [review]
Thanks. Should the title say "upgrade to 6/28" instead of 6/26?

File: cprover_bindings/src/irep/to_irep.rs (hunk 6 above)
Comment (jaisnan) [reply cprover_bindings/src/irep/to_irep.rs:264]: Yes, thank you

File: cprover_bindings/src/goto_program/expr.rs (hunk 2 above)
Comment (jaisnan) [reply cprover_bindings/src/goto_program/expr.rs:598]: I was trying to replicate the comment for `float_constant_from_bitpattern` and `double_constant_from_bitpattern` , which seems to be a C union equivalent for those types.
Should I remove these comments, or can I add a clarifying comment saying that these are C union representations?

File: cprover_bindings/src/irep/to_irep.rs (hunk 5 above)
Comment (jaisnan) [reply cprover_bindings/src/irep/to_irep.rs:736]: Added a comment to clarify, thanks for pointing it out.

File: cprover_bindings/src/goto_program/expr.rs (hunk 1 above)
Comment (tautschnig) [reply cprover_bindings/src/goto_program/expr.rs]: +1 for `Float16Constant`

File: cprover_bindings/src/goto_program/mod.rs [hunk 8]
```diff
@@ -6,7 +6,6 @@
 // There are a fair number of constructs in this module that are better maintained as
 // explicit pattern matching versus using the `matches!` macro.
 #![allow(clippy::match_like_matches_macro)]
-
```
Comment (tautschnig) [inline cprover_bindings/src/goto_program/mod.rs:9]: Unrelated change?

File: cprover_bindings/src/goto_program/typ.rs [hunk 9]
```diff
@@ -41,6 +41,10 @@ pub enum Type {
     FlexibleArray { typ: Box<Type> },
     /// `float`
     Float,
+    /// `Half float`
+    Float16,
+    /// `float 128`
```
Comment (tautschnig) [inline cprover_bindings/src/goto_program/typ.rs:46]: I don't think left ticks make sense with those (elsewhere there is C syntax between those tick marks). You could fix that by using `_Float16` and `_Float128`, which are the type names that will be supported by the next C standard.

File: cprover_bindings/src/goto_program/expr.rs (hunk 1 above)
Comment (jaisnan) [reply cprover_bindings/src/goto_program/expr.rs]: Updating the name to `Float16Constant`

File: cprover_bindings/src/irep/to_irep.rs [hunk 10]
```diff
@@ -695,6 +714,30 @@ impl ToIrep for Type {
...
+                // Sign bit: 1
+                named_sub: linear_map![
+                    (IrepId::F, Irep::just_int_id(10)),
+                    (IrepId::Width, Irep::just_int_id(16)),
+                    (IrepId::CCType, Irep::just_id(IrepId::Float16)),
+                ],
+            },
+            // Fraction bits: 112
+            // Precision bits: 52
+            // Sign bit: 1
```
Comment (tautschnig) [inline cprover_bindings/src/irep/to_irep.rs:731]: Can I suggest consistency in where you place the comment?

File: kani-compiler/src/kani_middle/attributes.rs [hunk 11]
```diff
@@ -1034,10 +1034,9 @@ fn attr_kind(tcx: TyCtxt, attr: &Attribute) -> Option<KaniAttributeKind> {
                     .intersperse("::")
                     .collect::<String>();
                 KaniAttributeKind::try_from(ident_str.as_str())
-                    .map_err(|err| {
+                    .inspect_err(|&err| {
                         debug!(?err, "attr_kind_failed");
                         tcx.dcx().span_err(attr.span, format!("unknown attribute `{ident_str}`"));
-                        err
                     })
```
Comment (tautschnig) [inline kani-compiler/src/kani_middle/attributes.rs:1040]: How is this change related to the others?

Comment (jaisnan) [reply kani-compiler/src/kani_middle/attributes.rs:1040]: Clippy enforced these, seems to be a part of the toolchain upgrade to clippy itself?

Comment (tautschnig) [reply kani-compiler/src/kani_middle/attributes.rs:1040]: Ok, thank you for clarifying!

File: cprover_bindings/src/goto_program/typ.rs [hunk 12]
```diff
@@ -166,6 +170,8 @@ impl DatatypeComponent {
             | Double
             | FlexibleArray { .. }
             | Float
+            | Float16
+            | Float128
```
Comment (tautschnig) [inline cprover_bindings/src/goto_program/typ.rs:174]: We have methods `is_float` and `is_double` - please create equivalents of those and inspect all use cases of `is_float` and `is_double` to see whether you have to add calls to the newly added methods.

Comment (jaisnan) [reply cprover_bindings/src/goto_program/typ.rs:174]: Added new methods for `f16` and `f128`

File: cprover_bindings/src/irep/to_irep.rs [hunk 13]
```diff
@@ -695,6 +714,30 @@ impl ToIrep for Type {
...
+                    (IrepId::F, Irep::just_int_id(10)),
+                    (IrepId::Width, Irep::just_int_id(16)),
+                    (IrepId::CCType, Irep::just_id(IrepId::Float16)),
+                ],
+            },
+            Type::Float128 => Irep {
+                id: IrepId::Floatbv,
+                sub: vec![],
+                // Fraction bits: 112
+                // Precision bits: 52
```
Comment (zhassan-aws) [inline cprover_bindings/src/irep/to_irep.rs:733]: I think this should be 15.

**Review by zhassan-aws — APPROVED** [review]

File: cprover_bindings/src/irep/to_irep.rs (hunk 13 above)
Comment (jaisnan) [reply cprover_bindings/src/irep/to_irep.rs:733]: Right, yes.

**Review by celinval — COMMENTED** [review]
Can you please add some tests?

**Comment by zhassan-aws** [comment]
> Can you please add some tests?

I was going to mention this in my overall comment, but completely forgot 🙈

**Review by celinval — DISMISSED** [review]
Can you at least extend this test to include the new floating points?

https://github.com/model-checking/kani/blob/main/tests/kani/FloatingPoint/main.rs

**Review by tautschnig — APPROVED** [review]

**Review by adpaco-aws — APPROVED** [review]
